packages/ui-buttons/src/ToggleButton/props.ts

```diff
@@ -37,90 +37,116 @@ import type {
   PositionConstraint,
   PositionMountNode
 } from '@instructure/ui-position'
+import type { ViewProps } from '@instructure/ui-view'
 
 type ToggleButtonOwnProps = {
-  screenReaderLabel: string
-  renderTooltipContent: React.ReactNode | ((...args: any[]) => any)
-  renderIcon: React.ReactNode | ((...args: any[]) => any)
-  status: 'pressed' | 'unpressed'
-  as?: AsElementType
-  interaction?: 'enabled' | 'disabled' | 'readonly'
-  size?: 'small' | 'medium' | 'large'
-  elementRef?: (element: Element | null) => void
-  onClick?: (...args: any[]) => any
-  color?: string
-  isShowingTooltip?: boolean
-  mountNode?: PositionMountNode
-  placement?: PlacementPropValues
-  constrain?: PositionConstraint
-}
-
-type PropKeys = keyof ToggleButtonOwnProps
-
-type AllowedPropKeys = Readonly<Array<PropKeys>>
-
-type ToggleButtonProps = ToggleButtonOwnProps &
-  OtherHTMLAttributes<ToggleButtonOwnProps>
-
-const propTypes: PropValidators<PropKeys> = {
   /**
    * Text to output only to screen readers
    */
-  screenReaderLabel: PropTypes.string.isRequired,
+  screenReaderLabel: string
+
   /**
    * Text to render in the tooltip shown on hover/focus
    */
-  renderTooltipContent: PropTypes.oneOfType([PropTypes.node, PropTypes.func])
-    .isRequired,
+  renderTooltipContent: React.ReactNode | ((...args: any[]) => any)
+
   /**
    * An icon or function that returns an icon
    */
-  renderIcon: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
+  renderIcon: React.ReactNode | (() => React.ReactNode)
+
   /**
    * Toggles the `aria-pressed` attribute on the button (`true` if `pressed`; `false` if `unpressed`)
    */
-  status: PropTypes.oneOf(['pressed', 'unpressed']).isRequired,
+  status: 'pressed' | 'unpressed'
+
   /**
    * The element to render as the component root; `button` by default
    */
-  as: PropTypes.elementType,
+  as?: AsElementType
+
   /**
    * Specifies if interaction with `ToggleButton` is `enabled`, `disabled`, or `readonly`
    */
-  interaction: PropTypes.oneOf(['enabled', 'disabled', 'readonly']),
+  interaction?: 'enabled' | 'disabled' | 'readonly'
+
   /**
    * The size of the `ToggleButton`
    */
-  size: PropTypes.oneOf(['small', 'medium', 'large']),
+  size?: 'small' | 'medium' | 'large'
+
   /**
    * Provides a reference to `ToggleButton`'s underlying HTML element
    */
-  elementRef: PropTypes.func,
+  elementRef?: (element: Element | null) => void
+
   /**
    * Callback fired when the `ToggleButton` is clicked
    */
-  onClick: PropTypes.func,
+  onClick?: (
+    event: React.KeyboardEvent<ViewProps> | React.MouseEvent<ViewProps>
+  ) => void
+
   /**
    * The color in which to display the icon
    */
-  color: PropTypes.string,
+  color?: 'primary' | 'primary-inverse' | 'secondary' | 'success' | 'danger'
```

Comment:
I specified the type for "color", because it used to be `string`, but it gets passed to IconButton, which has this type above. It shouldn't be breaking because if they used a wrong string before, the IconButton prop check caught it.